src/main/java/org/jenkinsci/plugins/github_branch_source/ForkPullRequestDiscoveryTrait.java

```diff
@@ -316,6 +318,94 @@ public boolean isApplicableToOrigin(@NonNull Class<? extends SCMHeadOrigin> orig
     }
 
     /**
+     * An {@link SCMHeadAuthority} that trusts contributors to the repository.
+     */
+    public static class TrustPermission
+            extends SCMHeadAuthority<GitHubSCMSourceRequest, PullRequestSCMHead, PullRequestSCMRevision> {
+        @NonNull
+        private final GHPermissionType permission;
+
+        /**
+         * Constructor.
+         */
+        @DataBoundConstructor
+        public TrustPermission(@NonNull String permission) {
+            GHPermissionType permissionType = GHPermissionType.ADMIN;
```

Comment:
I tend to agree with @jglick. For these purposes, `WRITE` is just as dangerous as `ADMIN`, so why do we care about `ADMIN`?